hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -589,6 +609,75 @@ public boolean mkdirs(final Path f, final FsPermission permission) throws IOExce
     }
   }
 
+  /**
+   * Checks for the entire parent hierarchy and returns if any directory exists and
+   * throws an exception if any file exists.
+   * @param path path to check the hierarchy for.
+   * @param tracingContext the tracingcontext.
+   * @throws IOException
+   */
+  private void checkParentChainForFile(Path path, TracingContext tracingContext) throws IOException {
+    if (directoryExists(path, tracingContext)) {
+      return;
+    }
+    for (Path current = path.getParent(), parent = current.getParent();
+         parent != null; // Stop when you get to the root
+         current = parent, parent = current.getParent()) {
+      if (directoryExists(current, tracingContext)) {
+        break;
+      }
+    }
+  }
+
+  /**
+   * Returns true if path is directory.
+   * @param path path to verify.
+   * @param tracingContext the tracingContext.
+   * @return true or false.
+   * @throws IOException
+   */
+  private boolean directoryExists(Path path, TracingContext tracingContext) throws IOException {
+    if (abfsStore.getListBlobs(path, path.toUri().getPath() + FORWARD_SLASH,
```

Comment:
let it be `getListBlobs(path, null, tracingContext, 2, true)`

Reply:
done